chef/cookbooks/neutron/recipes/common_agent.rb

```diff
@@ -257,10 +289,57 @@
         Chef::Log.info("#{name} usurped #{res[1].join(", ")} routes from #{bound_if}") unless res[1].empty?
       end
     end
+
+    execute "Neutron: add #{bound_if} to #{name}" do
+      command "ovs-vsctl add-port #{name} #{bound_if}"
+      not_if not_if_clause
+    end
     source = ::Nic.new(bound_if)
     # filter out cluster VIPs that a currently assigned to the interface
     addresses = source.addresses.select { |address| my_addresses.include? address }
     routes = source.routes
+
+    # Try to bring the cluster back up.
+    if handle_corosync_restart
+      service "start openais after network reconfiguration" do
+        service_name node[:corosync][:platform][:service_name]
+        action [:start]
+        not_if { addresses.empty? and routes.empty? }
```

Comment:
Right. I'll rework that to use ```not_if_clause``` here as well. I use ```addresses.empty? and routes.empty?```  here as an indicator for that we didn't reassign any interface or IP addresses. That should also indicate that corosync wasn't stopped and doesn't need to be restarted. (This should be true for all but the first chef-client run)

But this is also kind of a weak point that I didn't find a better solution for yet. If chef-client fails for what ever reason in between in ```"Have #{name} usurp config from #{bound_if}"``` block and here corosync will never be restarted on a subsequent run. I guess we could make the ```:start``` action unconditional here, it shouldn't do any harm, but I don't really like that either.